### ISSUE TITLE
Sentry v8 build issue using Carthage with static linkage

### ISSUE THREAD

**Issue by RomanTysiachnik** [getsentry/sentry-cocoa#2951]
### Platform

iOS

### Installed

Carthage

### Version

8.5.0

### Steps to Reproduce

Hello Sentry team,

We're trying to migrate from `7.31.5` to `8.x.x`, and we have a build issue preventing us from doing so (tried with 8.3.0, 8.4.0, and 8.5.0).

We use Carthage to manually build Sentry as xcframework applying some custom configuration parameters for it:
```
MACH_O_TYPE = staticlib
DEBUG_INFORMATION_FORMAT=dwarf
```

And here's how we configure Sentry:

```swift
SentrySDK.start { options in
    options.enableAutoPerformanceTracing = false
    options.enableAutoSessionTracking = false
    options.enableUIViewControllerTracing = false
    options.enableUserInteractionTracing = false
    options.enableAutoBreadcrumbTracking = false
    options.enableCaptureFailedRequests = false
    options.profilesSampleRate = 0
    options.enablePreWarmedAppStartTracing = true
}
```

**Static linkage** is vital for our product, and we can't switch to dynamic under any circumstances (at least for today) due to the importance of startup time.

### Expected Result

Compile and run.

### Actual Result

Build fails with the following issue:

<img width="467" alt="Screenshot 2023-04-24 at 17 25 16" src="https://user-images.githubusercontent.com/31652265/234046311-0c611019-5c41-4473-b9c0-f8578842a314.png">

**Comment by brustolin** [comment]
Hello @RomanTysiachnik thanks for letting us know. 
If I understood this right you download Sentry source code and compile it yourself using Carthage. 
You don't use our Carthage version right?

**Comment by RomanTysiachnik** [comment]
Hello @brustolin,

That's correct. That's how Carthage works, it checkouts the code and builds it into the framework. It is possible to specify a pre-built `binary`, but we prefer building on our own for safety and to control where we want to have static/dynamic linkage (e.g. static on iOS, dynamic on macOS).

**Comment by RomanTysiachnik** [comment]
@brustolin 

I feel like this issue might be related to [that report](https://developer.apple.com/forums/thread/727688). 

Since 8.x.x a `SentryPrivate.framework` was introduced and missing symbols are exactly from this framework.

**Comment by brustolin** [comment]
SentryPrivate should be embedded in SentrySDK, so, it dont look like is the same problem. But thanks for sharing that report.
I'll look into it.

**Comment by brustolin** [comment]
@RomanTysiachnik are you using `--use-xcframeworks` option with carthage?

Can you provide a small repro? Just a clean project targeting sentry with carthage, I cant reproduce your exact issue.

**Comment by RomanTysiachnik** [comment]
Hey @brustolin ,

Sorry for the long reply. I've identified the issue of why it is happening.

First of all, Sentry consists of 3 frameworks:
- _Sentry_
  - _SentryPrivate_
- _SentrySwiftUI_

Each of them has it's own `podspec` or mentioned as a `target` in Package.swift, so that the chosen package manager knows what dependencies it has.

The carthage, instead, uses **Xcode schemes**  to identify which targets it should build from the project and export them as `xcframework`. 

If we build `Sentry` as a `dynamic framework` using existing configuration, it will automatically embed static `SentryPrivate` inside.

However, if we build it as a `static framework`, it would build both of those frameworks, but only `Sentry` will be a part of `xcframework`, while it will expect that `SentryPrivate` will be linked by the resulting binary as:
- `Sentry.xcframework`
- `SentryPrivate.xcframework`

The reason for that is that a static framework can't embed anything inside of it - neither resources nor other frameworks.

So to achieve that, the Sentry project should have SentryPrivate exposed outside as an Xcode scheme, just like it is done with a podspec.

I've opened a merge request that makes exactly the following, and I've successfully tested that, achieving all those frameworks built as separate xcframeworks.
